include/zephyr/drivers/auxdisplay.h

```diff
@@ -0,0 +1,887 @@
+ *
+ * @param dev		Auxiliary display device instance
+ * @param brightness	The brightness level to set
+ *
+ * @retval		0 on success.
+ * @retval		-ENOSYS if not supported/implemented.
+ * @retval		-EINVAL if provided argument is invalid.
+ * @retval		-EIO if powered down.
+ * @retval		-errno Negative errno code on other failure.
+ */
+__syscall int auxdisplay_brightness_set(const struct device *dev,
+					uint8_t brightness);
+
+static inline int z_impl_auxdisplay_brightness_set(const struct device *dev,
+						   uint8_t brightness)
+{
+	struct auxdisplay_driver_api *api = (struct auxdisplay_driver_api *)dev->api;
+
+	if (!api->brightness_set) {
+		return -ENOSYS;
+	}
+
+#ifdef CONFIG_PM_DEVICE
+	enum pm_device_state state;
+
+	pm_device_state_get(dev, &state);
+
+	if (state == PM_DEVICE_STATE_SUSPENDED) {
+		return -EIO;
+	}
+#endif
+
+	return api->brightness_set(dev, brightness);
+}
+
+/**
+ * @brief		Get the backlight level details of an auxiliary display
+ *
+ * @param dev		Auxiliary display device instance
+ * @param backlight	Will be updated with the current backlight level
+ *
+ * @retval		0 on success.
+ * @retval		-ENOSYS if not supported/implemented.
+ * @retval		-errno Negative errno code on other failure.
+ */
+__syscall int auxdisplay_backlight_get(const struct device *dev,
+				       uint8_t *backlight);
+
+static inline int z_impl_auxdisplay_backlight_get(const struct device *dev,
+						  uint8_t *backlight)
+{
+	struct auxdisplay_driver_api *api = (struct auxdisplay_driver_api *)dev->api;
+
+	if (!api->backlight_get) {
+		return -ENOSYS;
+	}
+
+	return api->backlight_get(dev, backlight);
+}
+
+/**
+ * @brief		Update the backlight level of an auxiliary display
+ *
+ * @param dev		Auxiliary display device instance
+ * @param backlight	The backlight level to set
+ *
+ * @retval		0 on success.
+ * @retval		-ENOSYS if not supported/implemented.
+ * @retval		-EINVAL if provided argument is invalid.
+ * @retval		-EIO if powered down.
+ * @retval		-errno Negative errno code on other failure.
+ */
+__syscall int auxdisplay_backlight_set(const struct device *dev,
+				       uint8_t backlight);
+
+static inline int z_impl_auxdisplay_backlight_set(const struct device *dev,
+						  uint8_t backlight)
+{
+	struct auxdisplay_driver_api *api = (struct auxdisplay_driver_api *)dev->api;
+
+	if (!api->backlight_set) {
+		return -ENOSYS;
+	}
+
+#ifdef CONFIG_PM_DEVICE
+	enum pm_device_state state;
+
+	pm_device_state_get(dev, &state);
+
+	if (state == PM_DEVICE_STATE_SUSPENDED) {
+		return -EIO;
+	}
+#endif
+
+	return api->backlight_set(dev, backlight);
+}
+
+/**
+ * @brief	Check if an auxiliary display driver is busy
+ *
+ * @param dev	Auxiliary display device instance
+ *
+ * @retval	1 on success and display busy.
+ * @retval	0 on success and display not busy.
+ * @retval	-ENOSYS if not supported/implemented.
+ * @retval	-errno Negative errno code on other failure.
+ */
+__syscall int auxdisplay_is_busy(const struct device *dev);
```

Comment:
This just seems prone to race conditions, IMHO. Please require synchronization within the driver. If blocking is an issue, add a timeout, and check for isr context.

Reply:
The only driver that implements it is the noritake and all it does is read the state of a GPIO, I don't see what needs to be sync'd?